### PR TITLE
Add documentation webpage

### PR DESCRIPTION
**Pull request by danieltrt**
This PR introduces documentation for Piranha.
It's based on docusaurus v2

### CODE REVIEW

File: .github/workflows/deploy_docs.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          # Build output to publish to the `gh-pages` branch:
+          publish_dir: ./site/build
+          # The following lines assign commit authorship to the official
+          # GH-Actions bot for deploys to `gh-pages` branch:
+          # https://github.com/actions/checkout/issues/13#issuecomment-724415212
+          # The GH actions bot is used by default if you didn't specify the two fields.
+          # You can swap them out with your own user credentials.
+          user_name: github-actions[bot]
```
Comment (ketkarameya) [inline .github/workflows/deploy_docs.yml:40]: do we need this if this is default?

File: site/docs/reference/getting-started/stale.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
+---
```
Comment (ketkarameya) [inline site/docs/reference/getting-started/stale.md:1]: is this file stale? duplicate?

File: .github/workflows/deploy_docs.yml [hunk 3]
```diff
@@ -0,0 +1,41 @@
+name: Deploy to GitHub Pages
+
+on:
+  push:
+    branches:
+      - website
+    # Review gh actions docs if you want to further define triggers, paths, etc
+    # https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#on
```
Comment (ketkarameya) [inline .github/workflows/deploy_docs.yml:8]:
```suggestion

```

File: .github/workflows/deploy_docs.yml [hunk 4]
```diff
@@ -0,0 +1,41 @@
+name: Deploy to GitHub Pages
+
+on:
+  push:
+    branches:
+      - website
+    # Review gh actions docs if you want to further define triggers, paths, etc
```
Comment (ketkarameya) [inline .github/workflows/deploy_docs.yml:7]:
```suggestion

```

**Review by ketkarameya — APPROVED** [review]
some nits/comments/questions

File: site/docs/reference/getting-started/stale.md (hunk 2 above)
Comment (danieltrt) [reply site/docs/reference/getting-started/stale.md:1]: It's a file about stale flags :)

**Pull request merged**